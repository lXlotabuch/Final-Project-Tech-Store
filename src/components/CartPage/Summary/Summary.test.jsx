/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Router } from 'react-router-dom/cjs/react-router-dom.min';
import { createMemoryHistory } from 'history';
import { SummaryComponent } from './Summary';

describe('SummaryComponents', () => {
  test('render', () => {
    const {asFragment} = render(<SummaryComponent
      summary={156985}
    />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('handle go to checkout', () => {
    const onClick = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <SummaryComponent
          summary={156985}
          onClick={onClick}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /proceed to checkout/i}))
    expect(history.location.pathname).toBe('/checkout')
  })

  test('handle show message', () => {
    const onClick = jest.fn()
    const {getByText} = render(
      <SummaryComponent
        summary={0}
        onClick={onClick}
      />
    )
    fireEvent.click(screen.getByRole('button', {name: /proceed to checkout/i}))

    expect(getByText('Your shopping cart is empty. Please add items to your shopping cart')).toBeDefined()
  })
})