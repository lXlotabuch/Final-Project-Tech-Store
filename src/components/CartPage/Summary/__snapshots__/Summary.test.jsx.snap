// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryComponents render 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz glvBWj"
  >
    <h4>
      Summary
    </h4>
    <div
      class="sc-dlfnbm sc-gKsewC jcczqT kcTMhS"
    >
      <div>
        <p
          class="sumtotal"
        >
          Items total:
        </p>
      </div>
      <div>
        <span
          class="sumtotal"
        >
          156985â‚´
        </span>
      </div>
    </div>
    <div>
      <button
        class="ant-btn ant-btn-round sc-bdfBwQ bmYKhg"
        color="yellow"
        type="button"
      >
        <span>
          Proceed to Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
